### PR TITLE
Use open-mixtral-8x22b over mistral-medium

### PR DESCRIPTION
**Pull request by PopDaph**
## Description

Set `open-mixtral-8x22b` over `mistral-medium`.

## Risk

Break all mistral based agents.
Can be rolled back if we switch back all agents.

## Deploy Plan

- Deploy to Front
- Run `npx tsx ./scripts/update_assistants_models.ts --fromModel mistral-medium --toModel mistral-8x22b --execute`
- Prepare another PR to remove the config for mistral-medium.

### CODE REVIEW

**Review by flvndvd — COMMENTED** [review]
~We should first keep both models as it will break for users with assistant that have been migrated already.~

File: types/src/front/lib/assistant.ts [hunk 1]
```diff
@@ -204,6 +206,18 @@ export const MISTRAL_MEDIUM_MODEL_CONFIG: ModelConfigurationType = {
   shortDescription: "Mistral's smartest model.",
   supportsMultiActions: false,
 };
+export const MISTRAL_8X22B_MODEL_CONFIG: ModelConfigurationType = {
+  providerId: "mistral",
+  modelId: MISTRAL_8X22B_MODEL_ID,
+  displayName: "Mistral 8x22B",
+  contextSize: 64_000,
+  recommendedTopK: 16,
+  recommendedExhaustiveTopK: 56, // 28_672
```
Comment (flvndvd) [inline types/src/front/lib/assistant.ts:215]: Are we sure about this?

**Review by flvndvd — APPROVED** [review]
LGTM

File: types/src/front/lib/assistant.ts (hunk 1 above)
Comment (PopDaph) [reply types/src/front/lib/assistant.ts:215]: No, but I could not find the info on the doc :/

Comment (spolu) [reply types/src/front/lib/assistant.ts:215]: We can likely push it a bit? what are other 64k model using?

**Review by spolu — APPROVED** [review]

File: types/src/front/lib/assistant.ts (hunk 1 above)
Comment (PopDaph) [reply types/src/front/lib/assistant.ts]: I added a commit!

**Comment by PopDaph** [comment]
### Closing without a merge: 

**Why?**
I tested on Front Edge and got some "Error streaming tokens from Mistral AI": https://dust.tt/w/78bda07b39/a/6a27050429/runs/e4559970897d898062dd99fb22fb927e7d1d4d320ca4c519cfd4948c7932389e

**Investigation**
I duplicated locally assistant-v2-generator with the same input to see the valid error and I get a: 
```
Error querying `mistral:open-mixtral-8x22b`: error=Too many retries (1): [model_error(retryable=true)] "Tool call id has to be defined in serving mode."
```
The model does not support function_call and expect a tool_call_id.

**Next step**
I will not merge this PR and instead retire mistral-medium and migrate all agents based on mistral-medium to mistral large.

**Pull request closed without merging**